apps/ejabberd/src/ejabberd_c2s.erl

```diff
@@ -139,6 +140,9 @@ broadcast(FsmRef, Type, From, Packet) ->
 stop(FsmRef) ->
     ?GEN_FSM:send_event(FsmRef, closed).
 
+store_info(FsmRef, User, Server, Resource, KV) ->
+    FsmRef ! {store_info, User, Server, Resource, KV, self()}.
```

Comment:
In the end I left it as is - plain `!` instead of `?GEN_FSM:send_all_state_event`.